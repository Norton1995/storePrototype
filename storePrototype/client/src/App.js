import React, { Component } from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
//Import the page components
import Home from './pages/Home.jsx';
import Header from './pages/components/Header.jsx';
import Footer from './pages/components/Footer.jsx';
import Products from './pages/Products.jsx';
import FAQ from './pages/InfoPage.jsx';

class App extends Component {
    render() {
        return (
      <div>
      <Router>
        <div>
{/*Based on the URL path, display the following components, header and footer will be displayed at all
times as "/" will always be present on a path, using React Router*/}
          <Route path="/" component={Header}/>
          <Route exact path="/" component={Home}/>
          <Route path="/products" component={Products}/>
          <Route path="/about" component={FAQ}/>
          <Route path="/" component={Footer}/>
        </div>
      </Router>
    </div>
    );
  }
}

//Export app for use within project
export default App;
