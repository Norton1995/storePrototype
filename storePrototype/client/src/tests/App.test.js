import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('App renders correctly', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<App />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('should render correctly in "debug" mode', () => {
        const component = shallow(<App debug />);

        expect(component).toMatchSnapshot();
    });

    it('should render correctly with no props', () => {
        const component = shallow(<App/>);
        expect(component).toMatchSnapshot();
    });
});