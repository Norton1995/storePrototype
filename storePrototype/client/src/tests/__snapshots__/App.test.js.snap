// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`App renders correctly should render correctly with no props 1`] = `ShallowWrapper {}`;
